manifest-build-tools/application/manifest_generator.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+# Copyright 2015-2016, EMC, Inc.
+
+"""
+The script generate a new manifest for a new branch according to another manifest
+
+usage:
+./on-tools/manifest-build-tools/HWIMO-BUILD on-tools/manifest-build-tools/application/manifest_generator.py \
+--source-manifest build-manifest/rackhd-devel \
+--dest-manifest build-manifest/rackhd-release-1.2.5 \
+--branch branch/release-1.2.5 \
+--force \
+--publish \
+--publish-branch master \
+--git-credential https://github.com,GITHUB
+
+The required parameters: 
+source-manifest: The path of manifest. The new manifest is generated according to it.
```

Comment:
what is the source-file used for ?
in my mind, you can obtain all information  you need from git repos , no need the old manifest file ?